packages/holocron/src/externalRegistry.js

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2023 American Express Travel Related Services Company, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,either express
+ * or implied. See the License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Map as iMap, fromJS } from 'immutable';
+import semver from 'semver';
+
+let registeredExternals = iMap();
+let requiredModuleExternals = iMap();
```

Comment:
Do we need two maps for this? Can the data be combined into one?

Reply:
no, they are different data sets serving different cases. combining to one would make for a more complicated structure.